subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/switch_plan/SwitchPlanBottomSheetDialog.kt

```diff
@@ -167,17 +169,12 @@ class SwitchPlanBottomSheetDialog @AssistedInject constructor(
                 when (it) {
                     is CurrentPurchase.Success -> {
                         logcat { "Switch flow: Successfully switched plans" }
+                        lifecycleOwner.lifecycleScope.launch(dispatcherProvider.io()) {
+                            subscriptionSwitchWideEvent.onUIRefreshed()
+                        }
```

Comment:
Nit: Making `onUIRefreshed()` non-suspending and launching a coroutine inside that method would remove some of the clutter from the code that you're instrumenting.